packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { KeyboardEventHandler, useEffect, useState } from 'react'
+import {
+  Box,
+  List,
+  Paper,
+  Typography,
+  FormControl,
+  InputAdornment,
+  IconButton,
+} from '@mui/material'
+import { useTheme } from '@mui/material'
+import { ColorPartial } from '@mui/material/styles/createPalette'
+import { OutlinedInput } from '@mui/material'
+import { ArrowUpward } from '@mui/icons-material'
+import SynapseChatInteraction from './SynapseChatInteraction'
+import { SkeletonParagraph } from '../Skeleton'
+import { useCreateAgentSession } from 'src/synapse-queries/chat/useChat'
+import { AgentAccessLevel } from '@sage-bionetworks/synapse-types'
+
+export type SynapseChatProps = {
+  initialMessage: string
+  agentId?: string // if provided, use this agent
+}
+
+export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
+  initialMessage,
+  agentId,
+}) => {
+  // Create a new Agent Session on mount, add ability to initialize this component with a starting message from the user.
+  const [agentSessionId, setAgentSessionId] = useState('')
+  const { mutate: createAgentSession } = useCreateAgentSession({
+    onSuccess: newAgentSession => {
+      setAgentSessionId(newAgentSession.sessionId)
+    },
+  })
+  const theme = useTheme()
+  const [messages, setMessages] = useState([initialMessage])
+  const [currentMessage, setCurrentMessage] = useState('')
+
+  useEffect(() => {
+    // on mount, create a new agent session!
+    if (createAgentSession) {
+      createAgentSession({
+        agentAccessLevel: AgentAccessLevel.PUBLICLY_ACCESSIBLE,
+        agentId: agentId,
+      })
+    }
+  }, [createAgentSession])
+
+  const handleSendMessage = () => {
+    if (currentMessage.trim()) {
+      setMessages([...messages, currentMessage])
+      setCurrentMessage('')
+    }
+  }
+
+  const handleKeyDown: KeyboardEventHandler<
+    HTMLInputElement | HTMLTextAreaElement
+  > = event => {
+    if (event.key === 'Enter') {
```

Comment:
does this allow Shift+Enter to start a new line?

Reply:
Good call

Reply:
Storybook pops out to an iframe link, but I think this will work in the apps.